storage/pin/pin_test.go

```diff
@@ -0,0 +1,328 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package pin
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"io/ioutil"
+	"mime"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/ethersphere/swarm/api"
+	"github.com/ethersphere/swarm/chunk"
+	"github.com/ethersphere/swarm/storage"
+	"github.com/ethersphere/swarm/storage/feed"
+	"github.com/ethersphere/swarm/storage/localstore"
+	"github.com/ethersphere/swarm/testutil"
+)
+
+// Pin a RAW file and unpin it multiple times
+func TestPinRawUpload(t *testing.T) {
+
+	p, f := getPinApiAndFileStore(t)
+	data := testutil.RandomBytes(1, 10000)
+	hash := uploadFile(t, f, data, false)
+
+	// test pin and unpin
+	pinUnpinAndFailIfError(t, p, hex.EncodeToString(hash), 5, true)
+
+}
+
+// Pin a encrypted RAW file and unpin it multiple times
+func TestPinRawUploadEncrypted(t *testing.T) {
+
+	p, f := getPinApiAndFileStore(t)
+	data := testutil.RandomBytes(1, 10000)
+	hash := uploadFile(t, f, data, true)
+
+	// test pin and unpin
+	pinUnpinAndFailIfError(t, p, hex.EncodeToString(hash), 5, true)
+
+}
+
+// Pin a simple collection and unpin it multiple times
+func TestPinCollectionUpload(t *testing.T) {
+
+	p, f := getPinApiAndFileStore(t)
+	hash := uploadCollection(t, p, f, false)
+
+	// test pin and unpin
+	pinUnpinAndFailIfError(t, p, hex.EncodeToString(hash), 1, false)
+
+}
+
+// Pin a encrypted simple collection and unpin it multiple times
+func TestPinCollectionUploadEncrypted(t *testing.T) {
+
```

Comment:
remove ws... you get the point... go through the entire pr and get rid of excess whitespaces

Reply:
Done